<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>520</width>
    <height>722</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>80</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Trock Testing Platform</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>520</width>
     <height>700</height>
    </size>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QTabWidget" name="analysis_prep">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="quantitative_tab">
       <property name="layoutDirection">
        <enum>Qt::LeftToRight</enum>
       </property>
       <attribute name="title">
        <string>Quantitative Analysis</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_8">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Input CSV</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLineEdit" name="analysisInputCSVField"/>
          </item>
          <item>
           <widget class="QPushButton" name="analysisInputCSVButton">
            <property name="text">
             <string>Browse</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QFrame" name="frame_2">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_11">
           <item>
            <widget class="QFrame" name="frame">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_9">
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_6">
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_8">
                  <item>
                   <widget class="QLabel" name="analysisStrategyLabel">
                    <property name="styleSheet">
                     <string notr="true">qproperty-alignment: AlignCenter;</string>
                    </property>
                    <property name="text">
                     <string>Strategies</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_5">
                  <item>
                   <widget class="QLineEdit" name="analysisBrowseStrategyField"/>
                  </item>
                  <item>
                   <widget class="QPushButton" name="analysisBrowseStrategyButton">
                    <property name="text">
                     <string>Browse</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="analysisThresholdFrame">
                <item>
                 <widget class="QLabel" name="label_3">
                  <property name="text">
                   <string>Threshold</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="analysisThreshold">
                  <property name="decimals">
                   <number>4</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="analysisReturnsFrame">
                <item>
                 <widget class="QLabel" name="label_4">
                  <property name="text">
                   <string>Returns In Calculation</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="analysisReturns">
                  <property name="decimals">
                   <number>0</number>
                  </property>
                  <property name="maximum">
                   <double>100.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QFrame" name="analysisStrategyFrame">
                <layout class="QHBoxLayout" name="horizontalLayout_9">
                 <item>
                  <widget class="QPushButton" name="clearStrategyButton">
                   <property name="text">
                    <string>Clear Strategies</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="addStrategyButton">
                   <property name="text">
                    <string>Add Strategy</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QTableWidget" name="analysisStrategyList">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>460</width>
                  <height>107</height>
                 </size>
                </property>
                <column>
                 <property name="text">
                  <string>Strategy</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>Threshold</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>Returns in Calculation</string>
                 </property>
                 <property name="whatsThis">
                  <string/>
                 </property>
                 <property name="textAlignment">
                  <set>AlignHCenter|AlignVCenter|AlignCenter</set>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>CSV file</string>
                 </property>
                </column>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_10">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_11">
               <item>
                <widget class="QLabel" name="label_5">
                 <property name="styleSheet">
                  <string notr="true">qproperty-alignment: AlignCenter;</string>
                 </property>
                 <property name="text">
                  <string>Add New Date Range</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_12">
               <item>
                <widget class="QLabel" name="label_6">
                 <property name="text">
                  <string>Start Date</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDateEdit" name="analysisStartDate"/>
               </item>
               <item>
                <widget class="QLabel" name="label_7">
                 <property name="text">
                  <string>End Date</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDateEdit" name="analysisEndDate"/>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QFrame" name="analysisDateButtonFrame">
             <layout class="QHBoxLayout" name="horizontalLayout_10">
              <item>
               <widget class="QPushButton" name="analysisClearDateButton">
                <property name="text">
                 <string>Clear Date Ranges</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="analysisAddDateButton">
                <property name="text">
                 <string>Add Date Range</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QTableWidget" name="analysisListDate">
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <column>
              <property name="text">
               <string>Start Date</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>End Date</string>
              </property>
             </column>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="analysisExecuteButton">
          <property name="text">
           <string>Execute</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="run_tab">
       <property name="layoutDirection">
        <enum>Qt::LeftToRight</enum>
       </property>
       <attribute name="title">
        <string>Run Strategy Module</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QVBoxLayout" name="programLayout">
          <item>
           <widget class="QLabel" name="strategy_module_label">
            <property name="text">
             <string>Strategy Module</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="strategyModuleLayout">
            <item>
             <widget class="QLineEdit" name="strategy_module_location"/>
            </item>
            <item>
             <widget class="QPushButton" name="browseModule">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="text">
               <string>Browse</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="input_csv_label">
            <property name="text">
             <string>Input CSV</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="inputCSVLayout">
            <item>
             <widget class="QLineEdit" name="input_csv_location"/>
            </item>
            <item>
             <widget class="QPushButton" name="browseCSV">
              <property name="text">
               <string>Browse</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="inputcsv_valid">
            <property name="font">
             <font>
              <weight>50</weight>
              <bold>false</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">color: rgb(255, 0, 0);</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="parameters_label">
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>Parameters</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QGridLayout" name="_2">
            <property name="leftMargin">
             <number>1</number>
            </property>
            <item row="0" column="1">
             <widget class="QLabel" name="threshold_label">
              <property name="text">
               <string>Threshold</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QDoubleSpinBox" name="threshold">
              <property name="decimals">
               <number>4</number>
              </property>
              <property name="maximum">
               <double>1.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>1.000000000000000</double>
              </property>
              <property name="value">
               <double>0.000100000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="returnInCalculationLayout">
            <item>
             <widget class="QLabel" name="returnsInCalculation_label">
              <property name="text">
               <string>Returns in calculation</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="returnsInCalculation">
              <property name="locale">
               <locale language="English" country="UnitedKingdom"/>
              </property>
              <property name="maximum">
               <number>9999</number>
              </property>
              <property name="value">
               <number>5</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="dateLayout">
            <item>
             <widget class="QLabel" name="start_date_label">
              <property name="text">
               <string>Start Date</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDateEdit" name="start_date">
              <property name="displayFormat">
               <string>dd/MM/yyyy</string>
              </property>
              <property name="calendarPopup">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="end_date_label">
              <property name="text">
               <string>End Date</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDateEdit" name="end_date">
              <property name="displayFormat">
               <string>dd/MM/yyyy</string>
              </property>
              <property name="calendarPopup">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="date_valid">
            <property name="styleSheet">
             <string notr="true">color:red;</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QGridLayout" name="executeLayout">
            <property name="spacing">
             <number>5</number>
            </property>
            <item row="0" column="0" alignment="Qt::AlignLeft">
             <widget class="QLabel" name="execution_status">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>80</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QPushButton" name="execute_button">
              <property name="text">
               <string>Execute</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="analysis_tab">
       <attribute name="title">
        <string>Analyse Order CSV File</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="strategyModuleLayout_2">
          <item>
           <widget class="QLabel" name="outputcsv_label">
            <property name="text">
             <string>Output Order CSV File:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="output_csv_location"/>
          </item>
          <item>
           <widget class="QPushButton" name="browse_outputcsv">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>Browse</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="sizeConstraint">
           <enum>QLayout::SetMinimumSize</enum>
          </property>
          <item>
           <widget class="QLabel" name="output_csv_valid">
            <property name="autoFillBackground">
             <bool>false</bool>
            </property>
            <property name="styleSheet">
             <string notr="true">color: rgb(255, 0, 0);</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="loadorder_button">
          <property name="text">
           <string>Load Order CSV File</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="listWidget"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="analysis_error_msg">
            <property name="styleSheet">
             <string notr="true">color: rgb(255, 0, 0);</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QPushButton" name="selectall">
            <property name="text">
             <string>Select All</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="deselectall">
            <property name="text">
             <string>Deselect All</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="inverseSelection">
            <property name="text">
             <string>Inverse Selection</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="showanalysis_button">
          <property name="text">
           <string>Display Analysis</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="companylabel">
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Trock Pty Ltd. Version (v1.30). Web: &lt;a href=&quot;http://www.trock.net/&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;http://www.trock.net/&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="openExternalLinks">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
